### PR TITLE
Flash fixes for 1.10

### PR DESCRIPTION
**Pull request by mbolivar**
This PR includes two fixes from https://github.com/zephyrproject-rtos/zephyr/pull/5232 which, after reflection, I think should go into 1.10.

The first fixes flashing documentation for v2m_beetle. The second fixes flashing itself for all nrfjprog boards.

### CODE REVIEW

File: boards/arm/v2m_beetle/doc/v2m_beetle.rst [hunk 1]
```diff
@@ -237,7 +237,7 @@ the :ref:`hello_world` application.
 .. zephyr-app-commands::
    :zephyr-app: samples/hello_world
    :board: v2m_beetle
-   :goals: build flash
+   :goals: build
```
Comment (dbkinder) [inline boards/arm/v2m_beetle/doc/v2m_beetle.rst]: I don't see any instructions here that tell the reader how to flash?

**Review by dbkinder — DISMISSED** [review]

File: boards/arm/v2m_beetle/doc/v2m_beetle.rst (hunk 1 above)
Comment (mbolivar) [reply boards/arm/v2m_beetle/doc/v2m_beetle.rst]: They're in the very next paragraph.

The board does not support 'make flash'. The cmake conversion added that by mistake. That is the error which this patch is correcting.

Comment (mbolivar) [reply boards/arm/v2m_beetle/doc/v2m_beetle.rst]: Here is my commit which made the mistake of adding "flash", which was merged during the cmake conversion. Note "make flash" is not present in the original text.

```
commit f0c95919b06fa64c356cb1713b6e103724e922f3
Author: Marti Bolivar <marti@opensourcefoundries.com>
Date:   Tue Nov 7 22:44:42 2017 -0500

    doc: boards: arm: convert make to cmake
    
    Unify documentation formatting and use zephyr-app-commands where
    applicable.
    
    Signed-off-by: Marti Bolivar <marti@opensourcefoundries.com>

diff --git a/boards/arm/v2m_beetle/doc/v2m_beetle.rst b/boards/arm/v2m_beetle/doc/v2m_beetle.rst
index beba0b718..cf6dcdc08 100644
--- a/boards/arm/v2m_beetle/doc/v2m_beetle.rst
+++ b/boards/arm/v2m_beetle/doc/v2m_beetle.rst
@@ -231,20 +231,13 @@ to `CMSIS-DAP Website`_.
 Flashing an application to V2M Beetle
 -------------------------------------
 
-The sample application :ref:`hello_world` is being used in this tutorial:
+You can build applications in the usual way. Here is an example for
+the :ref:`hello_world` application.
 
-.. code-block:: console
-
-   $ZEPHYR_BASE/samples/hello_world
-
-To build the Zephyr kernel and application, enter:
-
-.. code-block:: console
-
-   $ cd $ZEPHYR_BASE
-   $ . zephyr-env.sh
-   $ cd $ZEPHYR_BASE/samples/hello_world/
-   $ make BOARD=v2m_beetle
+.. zephyr-app-commands::
+   :zephyr-app: samples/hello_world
+   :board: v2m_beetle
+   :goals: build flash
 
 Connect the V2M Beetle to your host computer using the USB port and you should
 see a USB connection which exposes a Mass Storage (MBED) and a USB Serial Port.
```

Comment (mbolivar) [reply boards/arm/v2m_beetle/doc/v2m_beetle.rst]: Just for reference, this is the link to our discussion in the PR for 1.11: https://github.com/zephyrproject-rtos/zephyr/pull/5232#discussion_r154474902

**Review by nashif — APPROVED** [review]

File: boards/arm/v2m_beetle/doc/v2m_beetle.rst (hunk 1 above)
Comment (dbkinder) [reply boards/arm/v2m_beetle/doc/v2m_beetle.rst]: Doh!

**Review by dbkinder — APPROVED** [review]

**Pull request merged**